geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStore.scala

```diff
@@ -58,6 +62,33 @@ class AccumuloDataStore(val connector: Connector, override val config: AccumuloD
   private val statsTable = GeoMesaFeatureIndex.formatSharedTableName(config.catalog, "stats")
   override val stats = new AccumuloGeoMesaStats(this, statsTable, config.generateStats)
 
+  // If on a secured cluster, create a thread to periodically renew Kerberos tgt
+  private val kerberosRenewerThread : Option[Thread] = if (UserGroupInformation.isSecurityEnabled) {
+    logger.info("Creating thread to periodically renew TGT from keytab")
+    Some(new Thread(new Runnable {
+      def run() {
+        while (true) {
+          try {
+            logger.info("Checking whether TGT needs renewing...")
+            UserGroupInformation.getCurrentUser.checkTGTAndReloginFromKeytab()
+          } catch {
+            case iox: IOException => logger.warn("Error checking and renewing tgt: " + iox.toString)
+          }
+
+          try {
+            // Sleep for 10 mins
+            Thread.sleep(10*60*1000)
```

Comment:
Hmm... I wonder if there is a quick way to get the remaining time on the lease.  If so, it might be cleaner to 'schedule' a call right before the expiry to try and renew then.